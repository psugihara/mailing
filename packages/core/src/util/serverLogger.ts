import chalk from "chalk";

const { DEBUG, NODE_ENV } = process.env;

const PREFIX = "mailing";

export function log(message?: any, ...optionalParams: any[]) {
  console.log(chalk.white(PREFIX), message, ...optionalParams);
}

export function error(message?: any, ...optionalParams: any[]) {
  // e.g. when there is a jest test like instrumentHtml.test.ts
  // that intentionally throws an error, don't log from a
  // try-catch that logs and re-raises the error
  if ("test" === NODE_ENV) return;
  console.error(chalk.red(PREFIX), message, ...optionalParams);
}

export function debug(message?: any, ...optionalParams: any[]) {
  if (DEBUG)
    console.info(chalk.yellowBright(PREFIX), message, ...optionalParams);
}

export function logPlain(message?: any, ...optionalParams: any[]) {
  console.log(message, ...optionalParams);
}
