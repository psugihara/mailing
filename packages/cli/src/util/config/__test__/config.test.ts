import { log } from "../../serverLogger";
import fsExtra from "fs-extra";
import {
  defaults,
  looksLikeTypescriptProject,
  writeDefaultConfigFile,
  setConfig,
  getConfig,
  getQuiet,
} from "..";

jest.mock("../../serverLogger");

describe("writeDefaultConfigFile", () => {
  beforeAll(() => {
    // IMPORTANT: config.defaults uses memoization, so you should expect that
    // for all the tests in this file, the defaults will be what are listed here
    const TEST_DEFAULTS = {
      emailsDir: "./emails",
      outDir: "./previews_html",
      port: 3883,
      quiet: false,
      scaffoldOnly: false,
      typescript: true,
    };

    jest.spyOn(fsExtra, "existsSync").mockImplementation((path) => {
      return path === "./tsconfig.json" ? true : false;
    });
    expect(looksLikeTypescriptProject()).toBe(true);
    expect(defaults()).toEqual(TEST_DEFAULTS);
  });

  it("writes mailing.config.json if it doesn't exist", () => {
    const defaultJsonString = `{
  "typescript": true,
  "emailsDir": "./emails",
  "outDir": "./previews_html"
}
`;

    const mockWriteFileSync = jest
      .spyOn(fsExtra, "writeFileSync")
      .mockImplementation(() => false);

    const mockExistsSync = jest
      .spyOn(fsExtra, "existsSync")
      .mockImplementation(() => false);

    writeDefaultConfigFile();

    expect(mockExistsSync).toHaveBeenCalled();
    expect(mockWriteFileSync).toHaveBeenCalledWith(
      "./mailing.config.json",
      defaultJsonString
    );
    expect(log).toMatchSnapshot();
  });

  it("does not write mailing.config.json if it exists", () => {
    // config file exists
    jest.spyOn(fsExtra, "existsSync").mockImplementation((_path) => true);

    const mockReadJSON = jest
      .spyOn(fsExtra, "readJSONSync")
      .mockImplementation(() => ({
        typescript: true,
        emailsDir: "./emails",
        outDir: "./previews_html",
      }));

    const mockWriteFileSync = jest
      .spyOn(fsExtra, "writeFileSync")
      .mockImplementation(() => false);

    writeDefaultConfigFile();
    expect(mockReadJSON).not.toHaveBeenCalled();
    expect(mockWriteFileSync).not.toHaveBeenCalled();
    expect(log).not.toHaveBeenCalled();
  });

  describe("getQuiet", () => {
    it("works if config is defined", () => {
      setConfig(undefined);
      expect(getConfig).toThrow();
      setConfig({ quiet: true, port: 3456, emailsDir: "emails" });
      expect(getConfig).not.toThrow();
      expect(getQuiet()).toBe(true);
      setConfig({ quiet: false, port: 3456, emailsDir: "emails" });
      expect(getQuiet()).toBe(false);
    });
    it("works even if config is undefined", () => {
      setConfig(undefined);
      expect(getConfig).toThrow();
      expect(getQuiet()).toBe(false);
    });
  });
});
