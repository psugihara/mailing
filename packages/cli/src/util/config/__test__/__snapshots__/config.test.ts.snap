// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`writeDefaultConfigFile writes mailing.config.json if it doesn't exist 1`] = `
[MockFunction] {
  "calls": [
    [
      "added mailing.config.json in your project with the following contents:
{
  "typescript": true,
  "emailsDir": "./emails",
  "outDir": "./previews_html"
}
",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
